File: abc/abc115/abc115_c.fs
// https://atcoder.jp/contests/abc115/submissions/16818144

let [| N; K |] = stdin.ReadLine().Split() |> Array.map int

let h =
    [| for i in [ 0..N - 1 ] do
        yield (stdin.ReadLine() |> int) |]
    |> Array.sort

[| 0..N - K |]
|> Array.map (fun i -> h.[i + K - 1] - h.[i])
|> Array.min
|> stdout.WriteLine

File: abc/abc115/abc115_c.cs
// https://atcoder.jp/contests/abc115/submissions/16817638

using System;
using System.Linq;

class Program
{
    static void Main()
    {
        var input = Console.ReadLine().Split().Select(int.Parse).ToArray();
        var N = input[0];
        var K = input[1];
        var h = new long[N];
        for (int i = 0; i < N; i++)
        {
            h[i] = int.Parse(Console.ReadLine());
        }
        Array.Sort(h);
        var result = h[N - 1];
        for (int i = 0; i < N - K + 1; i++)
        {
            if (result > h[i + K - 1] - h[i])
            {
                result = h[i + K - 1] - h[i];
            }
        }
        Console.WriteLine(result);
    }
}
